### DISCUSSION TITLE
just want to know hybird search is search first in whole collections and then filter results by params or filter first and only search in range of filtered collections

### DISCUSSION THREAD

**Discussion by kkpssr** [milvus-io/milvus#29164] [Q&A and General discussion]
just want to know hybird search is search first in whole collections and then filter results by params or filter first and only search in range of filtered collections

**Comment by yhmo** [comment]
Milvus manages data by segments, each segment has an independent index. When you call search(), the search engine searches the segments one by one, and get a topk result from each segment, combines the results into a final topk result.

For each segment, the internal steps of hybrid search:
1. use the filter to test all the items in this segment, create a "bitset" list, it is a list of 0,1. For more info, read this doc: https://milvus.io/blog/2022-2-14-bitset.md
2. put the bitset and index into the search engine, ignore the items that are marked as "0", only compute the items that marked as "1"
